activiti-steps/activiti-steps-assertions-api/src/main/java/org/activiti/steps/assertions/AwaitProcessInstanceAssertions.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.steps.assertions;
+
+import org.activiti.api.process.model.ProcessInstance;
+import org.activiti.steps.matchers.OperationScopeMatcher;
+import org.activiti.steps.matchers.ProcessResultMatcher;
+import org.activiti.steps.matchers.ProcessTaskMatcher;
+
+import static org.awaitility.Awaitility.await;
+
+public class AwaitProcessInstanceAssertions implements ProcessInstanceAssertions {
+
+    private ProcessInstanceAssertions processInstanceAssertions;
+
+    public AwaitProcessInstanceAssertions(ProcessInstanceAssertions processInstanceAssertions) {
+        this.processInstanceAssertions = processInstanceAssertions;
+    }
+
+    @Override
+    public ProcessInstanceAssertions expect(ProcessResultMatcher... matchers) {
```

Comment:
@erdemedeiros here naming again.. is this the one for "events"? should we call this expectEvents() ?

Reply:
Yes, It would be better to provide different names. It can be easily done for the one checking events, but not sure about the others. How to give different names without breaking the readability. For now, I think we can differentiate only events:

```
            .expect(processInstance().hasStatus(ProcessInstance.ProcessInstanceStatus.RUNNING),
                        processInstance().hasName("my-process-instance-name"),
                        processInstance().hasBusinessKey("my-business-key"))
             .expect(processInstance().hasTask("Task User1",
                                                  Task.TaskStatus.ASSIGNED,
                                                  withAssignee("user1")))
             .expectEvents(processInstance().hasBeenStarted(),
                        startEvent("StartEvent_1").hasBeenStarted(),
                        startEvent("StartEvent_1").hasBeenCompleted(),
                        sequenceFlow("SequenceFlow_1uccvwa").hasBeenTaken(),
                        task("Task User1").hasBeenCreated(),
                        task("Task User1").hasBeenAssigned()
                )```

Reply:
@salaboy I think I found a way of differentiating names (`expectEvents`, `expectFields` and `expect`):
```
.expectEvents(
                        processInstance()
                                .hasBeenStarted(),
                        startEvent("StartEvent_1")
                                .hasBeenCompleted(),
                        sequenceFlow("SequenceFlow_052072h")
                                .hasBeenTaken(),
                        taskWithName("Task Group 1")
                                .hasBeenCreated()

                )
                .expectFields(processInstance()
                                      .businessKey("my-business-key"),
                              processInstance()
                                      .name("my-process-instance-name"))
                .expect(processInstance()
                                .hasTask("Task Group 1",
                                         Task.TaskStatus.CREATED))
                .andReturn();
```

Reply:
👍